npm_and_yarn/lib/dependabot/npm_and_yarn/package_manager.rb

```diff
@@ -2,20 +2,55 @@
 # frozen_string_literal: true
 
 require "dependabot/shared_helpers"
+require "dependabot/npm_and_yarn/version_selector"
 
 module Dependabot
   module NpmAndYarn
     class PackageManager
+      extend T::Sig
+      extend T::Helpers
       def initialize(package_json, lockfiles:)
         @package_json = package_json
         @lockfiles = lockfiles
         @package_manager = package_json.fetch("packageManager", nil)
+        @engines = package_json.fetch("engines", nil)
       end
 
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/PerceivedComplexity
       def setup(name)
```

Comment:
can we please do a PR to add Sorbet typing to this file as we agreed some weeks ago (this can be a separate PR, cc: @kbukum1 )

Reply:
@abdulapopoola , ill create a separate PR for this. (i added sorbet for new functions already)